promql/query_logger.go

```diff
@@ -96,12 +96,14 @@ func getMMapedFile(filename string, filesize int, logger log.Logger) ([]byte, io
 
 	err = file.Truncate(int64(filesize))
 	if err != nil {
+		file.Close()
```

Comment:
Why not `defer file.Close()`?

Reply:
@SuperQ `file` is only closed in error cases in this function, why `defer` as you don't want to close the file if all went well?

Reply:
https://man7.org/linux/man-pages/man2/mmap.2.html
```
After the mmap() call has returned, the file descriptor, fd, can
       be closed immediately without invalidating the mapping.
```

Reply:
yes that could be tested/done in another PR.

Reply:
Agree not to mix topics in the PR :)

Reply:
Could this get another approval then?

Reply:
Approved.